public/javascripts/controllers/farmer-ctrl.js

```diff
@@ -154,15 +192,105 @@ angular.module('jasmic.controllers')
                 $scope.newFarm = !$scope.newFarm;
                 $scope.farm = {};
             };
+
+            $scope.editFarmLocation = function(arrayindex) {
+
+                $scope.editFarm = !$scope.editFarm;
+                $scope.farm = $scope.farmer.fr_farms[arrayindex];
+
+                $scope.editingFarm = arrayindex;//set the farm currently being edited
+
+            };
+
             $scope.newCommodityItem = function() {
                 $scope.newCommodity = !$scope.newCommodity;
             };
+
+            $scope.editCommodityItem = function(arrayindex) {
+
+                if (arrayindex == null) document.getElementById("commodities-table").style.visibility='visible'; //if cancel button was clicked
+
+                else document.getElementById("commodities-table").style.visibility='hidden';
+
+                $scope.editCommodity = !$scope.editCommodity;
+                $scope.commodity = $scope.commodities[arrayindex];
+                $scope.commodity.co_availability_date= moment().toDate();
+                $scope.commodity.co_until = moment().add(7, 'days').toDate();
+
+                $scope.editingCommodity = arrayindex;//sets the commodity currently being edited
+
+            };
+
+            $scope.updateCommodity = function() {
+
+                CommodityEditFactory.update({id:$scope.commodities[$scope.editingCommodity].fa_farmer._id, commodity_id:$scope.commodity._id}, $scope.commodity, function(success) {
+
+                    $window.scrollTo(0,0);
+
+                    showDialog($mdDialog, {statusText:"Successfully Updated!"}, false);
+
+                    $scope.editCommodity = false;
+
+                    document.getElementById("commodities-table").style.visibility='visible';
+
+                }, function (error) {
+
+                    $window.scrollTo(0,0);
+
+                    showDialog($mdDialog, {statusText:"Error Updating Commodity!"}, true);
+
+                });
+
+            }
+
+
+            $scope.updateFarm = function() {
+
+                $scope.editFarm = false;
+
+                FarmEditFactory.update({id:$scope.farmer._id, farm_id:$scope.farm._id}, $scope.farm, function(success) {
+
+                    $window.scrollTo(0,0);
+
+                    showDialog($mdDialog, {statusText:"Successfully Updated!"}, false);
+
+                }, function (error) {
+
+                    $window.scrollTo(0,0);
+
+                    showDialog($mdDialog, {statusText:"Error Updating Farm!"}, true);
+
+                });
+
+            }
+
+
             $scope.newCommodity = false;
             $scope.newFarm = false;
+            $scope.editCommodity = false;
 
 
             var selectedDistrict;
 
+
+            /*
+
+            This hides all the commodity cards to ensure that the one clicked is the only one on screen
+
+             */
+
+            $scope.hidePopupCommodities = function(commodity){
```

Comment:
@nick22891 please review this entire file and ensure that the angular framework is being used.